### DISCUSSION TITLE
Mixed environment

### DISCUSSION THREAD

**Discussion by whtkam** [sys4/netbox-plugin-dns#172] [Q&A]
Can the relationship between IPAM and the DNS plugin be made more flexible? I have the problem that I have IPs that are part of a DNS domain but also IP addresses that are not part of a DNS domain. When coupling is active, the empty DNS custom fields overwrite the DNS name and it is empty. One idea could be to have another boolean custom field so that you can selectively choose whether the IP address is linked to the DNS plugin or not. Maybe there is another or better solution for that

**Comment by peteeckel** [comment]
HI @whtkam, thanks for the idea. 

The IPAM coupling feature has been marked "experimental" because of exactly that kind of feature you're suggesting - we need to find out what's needed and where we potentially need to make some design decisions. And yes, your suggestion does make sense. So as soon as I find the time.

In what I currently think of as the final solution (or at least the first non-experimental) solution, some things are planned to work very differently from now. The idea I'm envisioning is to use `dns_name` as the input field and just map DNS views to IPAM prefixes. That way it will be possible to generate DNS entries for IPAM addresses in multiple zones, which is important in multi-view environments where an IP address may exist in multiple zones. Your suggested additional boolean field is, however, still necessary in that scenario, but it will not be sufficient - it's possible that we need to override in which views records need to be created on a per-address base.

I'm hoping that NetBox 4 will bring some enhancements that will make implementing this functionality easier. One thing already came with NetBox 3.7: Plugins can now define their own fields in core models [NetBox #14173](https://github.com/netbox-community/netbox/issues/14173). Unfortunately that's not enough as we can't actually use them in the core view, so there is another FR [NetBox 14732](https://github.com/netbox-community/netbox/issues/14732) for NetBox that, if accepted, will make it possible to get rid of the custom fields, which are rather inflexible in some aspects. Currently there doesn't seem to be much movement in that area, but any additional support for the FR will probably help.

There are some other enhancements in 4.0 that will hopefully make implementing this kind of interaction with plugins easier - I'm watching 4.0 progress and following up on new developments there closely.
